codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java

```diff
@@ -326,20 +328,164 @@ public final String serviceName() {
+            new ClientExecutionParams<StreamingOutputOperationRequest, StreamingOutputOperationResponse>()
+                .withMarshaller(new StreamingOutputOperationRequestMarshaller(protocolFactory))
+                .withResponseHandler(responseHandler).withErrorResponseHandler(errorResponseHandler)
+                .withInput(streamingOutputOperationRequest), asyncResponseHandler);
+    }
+
+    /**
+     * Some paginated operation with result_key in paginators.json file<br/>
+     * <p>
+     * This is a variant of
+     * {@link #paginatedOperationWithResultKey(software.amazon.awssdk.services.json.model.PaginatedOperationWithResultKeyRequest)}
+     * operation. The return type is a custom publisher that can be subscribed to request a stream of response pages.
+     * SDK will internally handle making service calls for you.
+     * </p>
+     * <p>
+     * When the operation is called, an instance of this class is returned. At this point, no service calls are made yet
+     * and so there is no guarantee that the request is valid. The subscribe method should be called as a request to
+     * stream data. For more info, see {@link org.reactivestreams.Publisher#subscribe(org.reactivestreams.Subscriber)}.
+     * If there are errors in your request, you will see the failures only after you start streaming the data.
+     * </p>
+     *
+     * <p>
+     * The following are few ways to use the response class:
+     * </p>
+     * 1) Using the forEach helper method. This uses @
+     * {@link software.amazon.awssdk.core.pagination.async.SequentialSubscriber} internally
+     *
+     * <pre>
+     * {@code
+     * software.amazon.awssdk.services.json.paginators.PaginatedOperationWithResultKeyPublisher publisher = client.paginatedOperationWithResultKeyPaginator(request);
+     * CompletableFuture<Void> future = publisher.forEach(res -> { // Do something with the response });
+     * future.get();
+     * }
+     * </pre>
+     *
+     * 2) Using a custom subscriber
+     *
+     * <pre>
+     * {@code
+     * software.amazon.awssdk.services.json.paginators.PaginatedOperationWithResultKeyPublisher publisher = client.paginatedOperationWithResultKeyPaginator(request);
+     * publisher.subscribe(new Subscriber<software.amazon.awssdk.services.json.model.PaginatedOperationWithResultKeyResponse>() {
+     *
+     * public void onSubscribe(org.reactivestreams.Subscriber subscription) { //... };
+     *
+     *
+     * public void onNext(software.amazon.awssdk.services.json.model.PaginatedOperationWithResultKeyResponse response) { //... };
+     * });}
+     * </pre>
+     *
+     * As the response is a publisher, it can work well with third party reactive streams implementations like RxJava2.
+     * <p>
+     * <b>Note: If you prefer to have control on service calls, use the
+     * {@link #paginatedOperationWithResultKey(software.amazon.awssdk.services.json.model.PaginatedOperationWithResultKeyRequest)}
+     * operation.</b>
+     * </p>
+     *
```

Comment:
As the body of the methods are empty, didn't think it would be useful to add all of them and clutter the docs. Anyways when customers tries to add code like this, compiler will give errors about the missing methods